### PR TITLE
Updated the "TFX Taxi Cab Classification Pipeline" sample

### PR DESCRIPTION
**Pull request by Ark-kun**
Modernized the sample pipeline code.
/assign @gaoning777 @hongye-sun
/label area/samples

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/1115)

### CODE REVIEW

**Comment by Ark-kun** [comment]
The sample tests have passed: https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/kubeflow_pipelines/1115/kubeflow-pipeline-sample-test/1115424345984864257

File: samples/tfx/taxi-cab-classification-pipeline.py [hunk 1]
```diff
@@ -14,128 +14,22 @@
...
-            '--server-name', tf_server_name
-        ]
-    )
+import kfp
+from kfp import components
+from kfp import dsl
+from kfp import gcp
+
+
+dataflow_tf_data_validation_op  = components.load_component_from_url('https://raw.githubusercontent.com/kubeflow/pipelines/deb4d0ed41662055031832f4ba03e324dd609143/components/dataflow/tfdv/component.yaml')
```
Comment (IronPan) [inline samples/tfx/taxi-cab-classification-pipeline.py]: i thought we are going to deprecate the intermediate yaml?
@hongye-sun @gaoning777 what is the plan?

Comment (Ark-kun) [reply samples/tfx/taxi-cab-classification-pipeline.py]: The fate of intermediate yaml (the data format passed to the backend) is currently unclear, but the re-usable components are here to stay.
Loading declarative component definition files provides better experience for the users compared to copy-pasting the code snippets between pipelines and notebooks and makes the pipeline code files much shorter. (Of course nobody prevents the user from doing things the old copy-paste way.)

The recently released AI-Hub showcases that experience - component authors publish components and the pipeline authors can just load and start using them. We now have more than 50 components (AI-Hub and this repo) and that number is growing. @hongye-sun is a big contributor to this success.

**Pull request merged**